### PR TITLE
Channel service channel script

### PR DESCRIPTION
**Pull request by Drummerboy444**
The `ChannelService_Channel` script is what will be called repeatedly by the caster of the channel. This script will call through to the relevant channel script at the time (start channel, channel or finish channel). The channel service itself checks each step if the channel should be cancelled and calls the right script from there.

The previous commit before the most recent one had some fake channel scripts linked up to the left mouse, you could check out this commit (`git checkout f5aa1a7` and then `git checkout channel_service_channel_script` when you're done) if you want to test it out.

### CODE REVIEW

**Comment by Drummerboy444** [comment]
[Create ChannelService_Channel();](https://trello.com/c/Kxrkn7qF/9-create-channelservicechannel)

File: objects/oAbilityManager/Create_0.gml [hunk 1]
```diff
@@ -1,25 +1,27 @@
...
-	ONESHOT,
+enum Enum_AbilityTypes {
+	ONE_SHOT,
 	CHANNEL
 }
 
-map_ability_to_type = ds_map_create();
-map_ability_to_type[? Enum_Abilities.SLASH] = [Enum_Ability_types.ONESHOT];
-map_ability_to_type[? Enum_Abilities.SHIELDBASH] = [Enum_Ability_types.ONESHOT];
+mat_abilityToType = ds_map_create();
```
Comment (nick-wilson95) [inline objects/oAbilityManager/Create_0.gml]: mat -> map

File: scripts/AbilityManager_GetAbilityType/AbilityManager_GetAbilityType.gml [hunk 2]
```diff
@@ -5,4 +5,4 @@
 #endregion
 var _enum_ability = argument[0];
 
-return oAbilityManager.map_ability_to_type[? _enum_ability];
\ No newline at end of file
+return oAbilityManager.mat_abilityToType[? _enum_ability];
```
Comment (nick-wilson95) [inline scripts/AbilityManager_GetAbilityType/AbilityManager_GetAbilityType.gml]: mat -> map

Comment (nick-wilson95) [reply scripts/AbilityManager_GetAbilityType/AbilityManager_GetAbilityType.gml]: Also, please solve merge conflicts

File: objects/oAbilityManager/Create_0.gml (hunk 1 above)
Comment (Drummerboy444) [reply objects/oAbilityManager/Create_0.gml]: good spot

File: scripts/AbilityManager_GetAbilityType/AbilityManager_GetAbilityType.gml (hunk 2 above)
Comment (Drummerboy444) [reply scripts/AbilityManager_GetAbilityType/AbilityManager_GetAbilityType.gml]: have resolved the conflicts

**Pull request merged**